### PR TITLE
ci: test all platforms on py3.10

### PR DESCRIPTION
**Pull request by branchvincent**
# Pull Request Check List

Update tests to use the released version of Python 3.10. The required checks may need to be updated to include the new version.

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

**Review by neersighted — CHANGES_REQUESTED** [review]
A couple thoughts here. `skip.yml` will need to be updated with the same matrix (it serves to mark our required checks as passing when they are skipped).

Also, I'd like to keep a single test with Poetry from `master` installing itself. What I would suggest doing is keeping both the `python-version` and `include` changes, but add a `(HEAD)` marker to the name of the `include`d job.

**Comment by neersighted** [comment]
Also, Python 3.11 alpha (in addition to master/HEAD on 3.10) would be good to add as experimental as well. I think a suffix variable should be used to separate out the jobs.

**Comment by branchvincent** [comment]
sounds good! thoughts on adding 3.10 to `.cirrus.yaml`?

https://github.com/python-poetry/poetry/blob/aaf8806227ac1dc63eac2bb0991b0c1525350986/.cirrus.yml#L10-L11

**Comment by neersighted** [comment]
> 
> 
> sounds good! thoughts on adding 3.10 to `.cirrus.yaml`?
> 
> https://github.com/python-poetry/poetry/blob/aaf8806227ac1dc63eac2bb0991b0c1525350986/.cirrus.yml#L10-L11

Go ahead! The `python310` package is in the repos now.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,20 +17,27 @@ on:
...
-            python-version: "3.10.0-alpha - 3.10.0"
+            python-version: "3.10"
             experimental: true
             bootstrap-args: "--git https://github.com/python-poetry/poetry.git"
+            suffix: (HEAD)
+          - os: Ubuntu
+            python-version: "3.11-dev"
+            experimental: true
+            bootstrap-args: "--git https://github.com/python-poetry/poetry.git"
+            suffix: (Python Dev)
```
Comment (neersighted) [inline .github/workflows/main.yml:40]:
```suggestion
            suffix: (HEAD)
```

`HEAD` should be sufficient here as the version is already `3.11-dev`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -17,20 +17,27 @@ on:
 
 jobs:
   tests:
-    name: ${{ matrix.os }} / ${{ matrix.python-version }}
+    name: ${{ matrix.os }} / ${{ matrix.python-version }} ${{ matrix.suffix }}
```
Comment (neersighted) [inline .github/workflows/main.yml:20]: I wonder if this will add a space to the job name/break the required checks.

**Review by neersighted — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (neersighted) [reply .github/workflows/main.yml:20]: It looks like this doesn't cause any issues. Cool!

Comment (branchvincent) [reply .github/workflows/main.yml:20]: i wondered that as well, but the jobs still show the `Required` annotation, and highlighting the space between e.g. `3.6 (pull_request)` shows only a single space which is the same as before this PR. maybe github actions trims trailing spaces?

**Review by neersighted — APPROVED** [review]

**Comment by branchvincent** [comment]
hmm, i'm not a fan of the overall actions status showing a failure ❌  when it's only a failure from an experimental test...

**Comment by neersighted** [comment]
> 
> 
> hmm, i'm not a fan of the overall actions status showing a failure ❌ when it's only a failure from an experimental test...

Yeah, I've spent the last little while looking for a workaround, but there is none. Let's disable it for now until we can get Python 3.11 passing for a decent amount of the time.

**Comment by branchvincent** [comment]
[from this comment](https://github.com/actions/toolkit/issues/399#issuecomment-695039432), seems adding `continue-on-error` to the individual step will give the run a ✅ even when it fails. the downside would be manually inspecting the logs to determine the actual pass/fail status, but does give us an overall ✅

**Comment by neersighted** [comment]
I went ahead and merged this as I found a (test suite level) workaround for the broken Python 3.11 check. I'll create a PR that includes that and a couple other workflow enhancements.

**Pull request merged**